EIPS/eip-4883.md

```diff
@@ -0,0 +1,66 @@
+---
+eip: 4883
+title: Composable SVG NFT
+description: Compose an onchain SVG NFT by combining the SVG body of other onchain SVG NFTs
+author: Andrew B Coathup (@abcoathup), Alex (@AlexPartyPanda), Damian Martinelli (@damianmarti), blockdev (@0xbok), Austin Griffith (@austintgriffith)
+discussions-to: https://ethereum-magicians.org/t/eip-4883-composable-svg-nft/8765
+status: Draft
+type: Standards Track
+category: ERC
+created: 2022-03-08
+requires: 721
+---
+
+## Abstract
+Compose an onchain SVG NFT by combining the SVG body of one or more onchain SVG NFTs.  Render the SVG body as a string that can be concatenated.  
+
+## Motivation
+Onchain SVG NFTs allow for NFTs to be entirely onchain by returning artwork as SVG in a data URI of the tokenUri function.    
+Composability allows onchain SVG NFTs to be crafted. e.g. adding glasses or hats to a profile pic NFT
+
+## Specification
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+  
+Every ERC-4883 compliant contract must implement the `ERC4883` and `ERC165` interfaces, with `zIndex` being optional.
+```solidity
+/// @title ERC-4883 Non-Fungible Token Standard
+/// @dev See https://eips.ethereum.org/EIPS/eip-721
+interface IERC4883 is IERC165 {
+    function zIndex() external view returns (int256); // [OPTIONAL]
+    
+    function width() external view returns (uint256); // [OPTIONAL]
+    function height() external view returns (uint256); // [OPTIONAL]
+
+    function renderTokenById(uint256 id) external view returns (string memory);
+}
+```
+
+If the caller of the render function composes an SVG NFT they must have rights to use the SVG body in a composable SVG.  The token holder could optionally hold both the composer and composee NFTs or the token holder could hold the composer NFT, and the composer NFT could hold the composee NFT.  
+
+## Rationale
+  
+SVG uses a "painters model" of rendering.
+
+**Scalable Vector Graphics (SVG) 1.1 (Second Edition)**, section: **3.3 Rendering Order**
+>Elements in an SVG document fragment have an implicit drawing order, with the first elements in the SVG document fragment getting "painted" first. Subsequent elements are painted on top of previously painted elements.
+  
+### Render
+
+Render must return either an empty string or valid SVG element(s).
```

Comment:
This should be included in the "Specification" section.

The Specification section says "what" is proposed by the EIP, and Rationale says "why" those particular changes were chosen.